File: Neetcode 150/Arrays and Hashing/group-anagrams.cpp

// https://leetcode.com/problems/group-anagrams/description/

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        int size = strs.size();
        vector<vector<string>> ans;
        if(size == 0) return ans;

        unordered_map<string, vector<string>> mp;
        for(auto str : strs){
            string original = str;
            sort(str.begin(), str.end());
            mp[str].push_back(original);
        }

        for(auto it : mp) ans.push_back(it.second);

        return ans;
    }
};

File: Neetcode-150-js/Arrays and Hashing/group-anagrams.js

// https://leetcode.com/problems/group-anagrams/description/

/**
 * @param {string[]} strs
 * @return {string[][]}
 */

function sortString(str){
    return str.split("").sort().join("");
}

var groupAnagrams = function(strs) {
    const map = {};
    for(const str of strs){
        let sortedStr = sortString(str);
        if(map[sortedStr]){
            map[sortedStr].push(str);
        }else map[sortedStr] = [str];
    }

    const response = [];
    for(const arr of Object.values(map)){
      response.push(arr);
    }
    return response;
};